src/shared-components/checkbox/index.js

```diff
@@ -1,45 +1,47 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import SelectorButton from '../selectorButton';
+import { SelectorButton } from '../selectorButton';
 
 const propTypes = {
   checked: PropTypes.bool.isRequired,
   disabled: PropTypes.bool,
   onClick: PropTypes.func,
+  /**
+   * Text label displayed next to the checkbox
+   */
   children: PropTypes.oneOfType([
     PropTypes.arrayOf(PropTypes.node),
     PropTypes.node,
   ]),
   type: PropTypes.oneOf(['primary', 'secondary']),
+  /**
+   * Icons are only displayed at the large size, and inside the radio button
+   */
   icon: PropTypes.node,
   size: PropTypes.oneOf(['large', 'small']),
 };
 
-const defaultProps = {
-  children: null,
-  disabled: false,
-  icon: undefined,
-  onClick: () => undefined,
-  type: 'primary',
-  size: 'small',
-};
-
-const Checkbox = ({
+/**
+ * Checkboxs should be used to provide valuable information or additional context on a page. One of the best examples of a Checkbox is for product recommendations.
+ *
+ * `<Checkbox />` is a controlled component that represents a checkbox selection. This means that the `onClick` function should be used to change the checked state of the checkbox. Note that a group of checkbox buttons must be composed by a parent component.
+ */
+export const Checkbox = ({
   checked,
-  disabled,
-  onClick,
-  children,
-  type,
-  icon,
-  size,
+  children = null,
+  disabled = false,
+  icon = undefined,
+  onClick = undefined,
```

Comment:
Some files not yet converted to JS require this because TS picks up that we're typing this as a function, but if we do not say it is undefined this way there is a TS error about the prop not being provided. Confusing, but #423 will remove this type of error from `radiance-ui` anyway, so it is temporary.

Reply:
Im gonna merge my final TS convertion PR that covers alll components converted to TS then solve all merge conflicts for this PR so Reviewers can ignore components in JS for now if they want